workspaces/rbac/plugins/rbac-backend/src/service/enforcer-delegate.ts

```diff
@@ -512,26 +513,39 @@ export class EnforcerDelegate implements RoleEventEmitter<RoleEvents> {
     action: string,
     roles: string[],
   ): Promise<boolean> {
-    const filter = [];
+    const tempEnforcer = new Enforcer();
+    const model = newModelFromString(MODEL);
+
+    // copy filtered policies from enforcer to tempEnforcer
+    // model.addPolicies('p', 'p', [['role:admin', 'data:resource', 'read', 'allow']]);
+    let policies: string[][] = [];
     if (roles.length > 0) {
-      roles.forEach(role => {
-        filter.push({ ptype: 'p', v0: role, v1: resourceType, v2: action });
-      });
+      for (const role of roles) {
+        const filteredRolePolicies = await this.enforcer.getFilteredPolicy(
+          0,
+          ...[role, resourceType, action],
+        );
+        policies.push(...filteredRolePolicies);
+      }
     } else {
-      filter.push({ ptype: 'p', v1: resourceType, v2: action });
+      const regex = /\b(?:user|group):[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+\b/g;
+
+      const enforcePolicies = await this.enforcer.getFilteredPolicy(
+        1,
+        ...[resourceType, action],
+      );
+
+      policies = enforcePolicies.filter(policy => policy[0].match(regex));
```

Comment:
I believe that regular expressions can be resource-intensive in terms of performance. This code should be faster:

```
        policies = enforcePolicies.filter(
          policy =>
            policy[0].startsWith('user:') || policy[0].startsWith('group:'),
        );
```